M2/Macaulay2/packages/KGAlgebras.m2

```diff
@@ -0,0 +1,191 @@
+            M1 := matrix apply(group G, z-> z*y); --multiplies all group elements by the random element, giving column vectors, and we make a matrix M1 from those vectors
+            if rank M1 != rank ((group G)#0) then (
+                return image M1;
+            );--Checks that the module generated by M1 is not a strict submodule of KG
+            
+            H := apply(group G, transpose); --Generate list of group elements' matrices but transposed
+            MPP := transpose MP;
+            NP := kernel MPP;
+            yp := (gens NP)_0;
+            M2 := matrix apply(H,z -> z*yp);
+            if debugLevel == 1 then (
+                print(rank M2);
+            );
+            if rank M2 != rank (H#0) then (
+                return image M2;
+            );
+            if dim N == (degree(P))#0 then (
+                                return image M2;
+                        );
+                        
+        );
+    );
+);
+
+-- symmetric groups
+SnGroup = method(Options => {"K" => QQ})
+SnGroup (ZZ) := opts -> (n) -> (
+    xx := symbol xx;
+    R := (opts#"K")[xx_1 .. xx_n]; -- polynomial ring to do the group action on
+    cycle := new Array from 1 .. n; -- an n-cycle
+        finiteAction({permutationMatrix(n, cycle), -- the cycle
+            permutationMatrix(n, [1,2])}, -- a transposition
+            R)
+);
+
+
+
+
+
+
+beginDocumentation()
+
+doc ///
+    Key
+        KGAlgebras
+    Headline
+        A package for group rings
+    Description
+        Text
+            Implements group rings and the meataxe algorithm to compute the irreducibility of a finite dimensional representation over a finite field.
+
+///
+doc ///
+    Key
+        groupRing
+    Headline
+        Constructor for GroupRing
+    Usage
+        groupRing(G,K)
+    Inputs
+        G:
+            A FiniteGroupAction specifying a group.
+        K:
+            A field. 
+    Outputs
+        :
+            A group ring defined by group G and field K. 
+    Description
+        Text
+            Takes a FiniteGroupAction G and a field K and creates a group ring from them. 
+        Example
+            R = QQ[x1,x2,x3];
+            L = {matrix {{0,1,0},{1,0,0},{0,0,1}}, matrix {{0,0,1},{0,1,0},{1,0,0}}, matrix {{1,0,0},{0,0,1},{0,1,0}} };
+            G = finiteAction(L,R);
+            groupRing(G,QQ)
+///
+doc ///
+    Key
+        meataxe        
+    Headline
+        Function implementing the meataxe algorithm determining irreducibility of a finite dimensional finite group representation.
+    Usage
+        T = meataxe(G,S,K)
+    Inputs
+        G:FiniteGroupAction
+            A finite group action
+        S:FreeAlgebraQuotient
+            A KG-algebra
+        K:Ring
+            The underlying field of the other parameters.
+    Outputs
+        T:Boolean
+            Returns true if the Meataxe algorithm states that the K[G]-module is irreducible. Returns a nontrivial submodule if the module is reducible. 
```

Comment:
The documentation should be updated since this doesn't return a boolean any more.

Also, now that a module is return in both cases, how do we know if it's irreducible?